pkg/kubecost/allocation.go

```diff
@@ -2261,7 +2261,70 @@ func (asr *AllocationSetRange) Accumulate() (*AllocationSet, error) {
 }
 
 // TODO accumulate into lower-resolution chunks of the given resolution
-// func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) *AllocationSetRange
+func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
+	allocSetRange := &AllocationSetRange{allocations: []*AllocationSet{}}
+	var allocSet *AllocationSet
+	var err error
+
+	asr.Lock()
+	defer asr.Unlock()
+
+	// use window() with no lock
+	if asr.window().IsEmpty() {
+		return asr, nil
+	}
+
+	// Call total accumulate func if resolution is greater than total window duration
+	if resolution > asr.window().Duration() {
+		// unable to acquire lock here if old asr.accumulate is called
+		for _, as := range asr.allocations {
+			allocSet, err = allocSet.accumulate(as)
+			if err != nil {
+				return nil, err
+			}
+		}
+		allocSetRange.allocations = append(allocSetRange.allocations, allocSet)
+		return allocSetRange, nil
```

Comment:
This is basically a reimplementation of `(*AllocationSetRange).Accumulate()`, right? I think this is okay, but does the standard logic you've written for `AccumulateBy` not work in the `resolution > asr.window().Duration()` case?

Reply:
yep, agreed this was missed after some changes. Removed